micronota/db/tigrfam.py

```diff
@@ -0,0 +1,236 @@
+   GS  Gene symbol - can be applied automatically for prokaryotic
+       sequences.
+   EC  Enzyme Commission number. In the format  6.1.1.7  without the EC.
+       This field may contain more than one EC number, with a single space as
+       the separator.
+   EN  Expanded name - A fuller or informative alternate version of the
+       definition line. This compares (or is identical) to the more terse DE
+       definition that is used for automated annotation.
+   TP  Always "TIGRFAMs", the identifier of this database
+   === ========================================================================
+
+Proteins that score
+above the trusted cutoffs are believed to reside within the family and
+those falling below the noise cutoffs are believed to reside outside the
+family.  The margin of error with respect to presence or absence of a
+protein within a TIGRFAMs family is represented by the score range
+between noise and trusted cutoffs.
+
+Because the number of completed and nearly completed genomes
+has now entered the thousands, protein families are becoming
+very large, and exceptions may be found in certain equivalog
+families.  Exceptions usually represent neofunctionalizations
+that arise within an equivalog family, although some may
+represent paralogs that are minimally derived since their
+branching from the equivalog familiy (i.e. short branch length).
+
+TIGRFAMs now includes 57 models of type "exception",  a type
+of model that overrules annotation from an equivalog model,
+either to give more specific information or to correct annotation
+for a neofunctionalized subgroup.
+
+TIGRFAMS_GO_LINK
+^^^^^^^^^^^^^^^^
+Gene Ontology (GO) term assignments.
+Models may have  0, 1, or several GO assignments.
+
+
+Reference
+---------
+.. [1] http://www.ncbi.nlm.nih.gov/pubmed/12520025
+.. [2] http://www.ncbi.nlm.nih.gov/pubmed/23197656
+
+'''
+
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import shutil
+import gzip
+import tarfile
+from tempfile import mkdtemp
+from os.path import join
+from ftplib import FTP
+from sqlite3 import connect
+
+from ..bfillings.hmmer import hmmpress_hmm
+
+
+def prepare_db(out_d, prefix='tigrfam_v15.0'):
+    '''Download and prepare TIGRFAM database.
+
+    Parameters
+    ----------
+    out_d : str
+        The directory of output files.
+    prefix : str
+        The file name (without extensions) of the output files.
+    '''
+    server = 'ftp.tigr.org'
+    path = 'pub/data/TIGRFAMs'
+
+    hmm = 'TIGRFAMs_15.0_HMM.LIB.gz'
+    hmm_out = join(out_d, '%s.hmm' % prefix)
+
+    metadata = 'TIGRFAMs_15.0_INFO.tar.gz'
```

Comment:
that's a good idea. the only concern I have is that future update may also change how their files are organized, in which case this piece code will likely be changed anyway.